src/pgmoneta_ext/lib.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2024 The pgmoneta community
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list
+ * of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this
+ * list of conditions and the following disclaimer in the documentation and/or other
+ * materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without specific
+ * prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+ * OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+/* pgmoneta */
+#include <pgmoneta_ext.h>
+
+/* PostgreSQL */
+#include "postgres.h"
+#include "fmgr.h"
+
+PG_MODULE_MAGIC;
+
+PG_FUNCTION_INFO_V1(pgmoneta_version_ext);
+
+Datum
+pgmoneta_version_ext(PG_FUNCTION_ARGS)
+{
+   Datum version;
+   char v[1024];
+
+   memset(&v, 0, sizeof(v));
+   snprintf(&v[0], sizeof(v), "%s", VERSION);
+
+   version = CStringGetDatum(v);
```

Comment:
Can you try CStringGetTextDatum instead?

Reply:
I have tried it but encountered an error.
<img width="1263" alt="Screenshot 2024-05-18 at 6 20 34 AM" src="https://github.com/pgmoneta/pgmoneta_ext/assets/113565433/4bd16980-f1b2-48d7-9c46-863e8ea5f434">

Reply:
Try `#include "utils/builtins.h"`

Reply:
The compilation was successful, but the error persists.
<img width="461" alt="Screenshot 2024-05-18 at 6 38 24 AM" src="https://github.com/pgmoneta/pgmoneta_ext/assets/113565433/cc68adfc-b0e2-46af-99e8-b7bd59e9a48f">

Reply:
Hmm this is odd, perhaps just try returning a random string and see if it truncates the first character still. Also try `SELECT * from pgmoneta_version_ext()` as well, just in case.

Reply:
Are you sure that you have all the `include/` directories in the compile path ? 

```
make VERBOSE=1
```

is your friend

Reply:
I will try to figure it out.

Reply:
I figured it out—we just need to replace `CStringGetDatum` with `CStringGetTextDatum`, and it will work. The issue before was that I hadn't restarted the `psql`.